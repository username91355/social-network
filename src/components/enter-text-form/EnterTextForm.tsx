import React, {ChangeEvent} from 'react'
import styles from './EnterTextForm.module.css'
import {Button, Form} from 'antd'
import TextArea from 'antd/es/input/TextArea'

interface IProps {
    value: string
    title: string
    label?: string
    onChange: (value: string) => void
    send: () => void
    buttonDisabled?: boolean
}

export const EnterTextForm: React.FC<IProps> = React.memo(props => {

    const {value, title, label = null, onChange, send, buttonDisabled = false} = props

    const changeTextArea = (e: ChangeEvent<HTMLTextAreaElement>) => {
        onChange(e.currentTarget.value)
    }

    return (
        <div className={styles.textForm__wrapper}>
            <h3 className={styles.textForm__title}>{label}</h3>
            <Form>
                <Form.Item>
                    <TextArea className={styles.textForm__textArea} rows={4} onChange={changeTextArea} value={value}/>
                </Form.Item>
                <Form.Item>
                    <Button className={styles.textForm__button}
                            htmlType='submit'
                            disabled={buttonDisabled}
                            onClick={send}
                            type='primary'>
                        {title}
                    </Button>
                </Form.Item>
            </Form>
        </div>
    )
})