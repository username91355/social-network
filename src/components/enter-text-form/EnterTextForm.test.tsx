import {render, screen} from '@testing-library/react'
import {EnterTextForm} from './EnterTextForm'
import userEvent from "@testing-library/user-event";

const sendFn = jest.fn()
const changeFn = jest.fn()

describe('EnterTextForm test`s', () => {
    beforeAll(() => {
        window.matchMedia = window.matchMedia || function() {
            return {
                matches: false,
                addListener: jest.fn(),
                removeListener: jest.fn()
            }
        }
    });
    it('should be render with title, value and label', () => {
        render(<EnterTextForm title={'title'}
                              value={'value'}
                              label={'label'}
                              send={sendFn}
                              onChange={changeFn}/>)

        expect(screen.getByText('label')).toBeInTheDocument()
        expect(screen.getByText('value')).toBeInTheDocument()
        expect(screen.getByText('title')).toBeInTheDocument()
    })
    it('Change message text area should be work', () => {
        render(<EnterTextForm title={'title'}
                              value={'value'}
                              label={'label'}
                              send={sendFn}
                              onChange={changeFn}/>)

        userEvent.type(screen.getByRole('textbox'), 'test')

        expect(changeFn).toHaveBeenCalledTimes(4)
    })
    it('Send message should be work', () => {
        render(<EnterTextForm title={'title'}
                              value={'value'}
                              label={'label'}
                              send={sendFn}
                              onChange={changeFn}/>)

        userEvent.click(screen.getByRole('button'))

        expect(sendFn).toHaveBeenCalledTimes(1)
    })
    it('Enter text form span snapshot', () => {
        const element = render(<EnterTextForm title={'title'}
                                              value={'value'}
                                              label={'label'}
                                              send={sendFn}
                                              onChange={changeFn}
        />)

        expect(element).toMatchSnapshot()
    })
})