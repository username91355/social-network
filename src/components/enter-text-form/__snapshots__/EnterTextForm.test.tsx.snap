// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterTextForm test\`s Enter text form span snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="textForm__wrapper"
      >
        <h3
          class="textForm__title"
        >
          label
        </h3>
        <form
          class="ant-form ant-form-horizontal"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <textarea
                    class="ant-input textForm__textArea"
                    rows="4"
                  >
                    value
                  </textarea>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    class="ant-btn ant-btn-primary textForm__button"
                    type="submit"
                  >
                    <span>
                      title
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="textForm__wrapper"
    >
      <h3
        class="textForm__title"
      >
        label
      </h3>
      <form
        class="ant-form ant-form-horizontal"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <textarea
                  class="ant-input textForm__textArea"
                  rows="4"
                >
                  value
                </textarea>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  class="ant-btn ant-btn-primary textForm__button"
                  type="submit"
                >
                  <span>
                    title
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
